File: tests/model_discrepancy/model_discrepancy_synthetic_test_with_hyperparam_auto_1D/Assemble_Mass_and_Stiffness.m
function [M,S] = Assemble_Mass_and_Stiffness(m)
x = linspace(0, 1, m)';
h = x(2) - x(1);
M = diag(4 * ones(1, m)) + diag(ones(1, m - 1), 1) + diag(ones(1, m - 1), -1);
M(1, 1) = .5 * M(1, 1);
M(end, end) = .5 * M(end, end);
M = (1 / 6) * h * M;
S = diag(2 * ones(1, m)) + (-1) * diag(ones(1, m - 1), 1) + (-1) * diag(ones(1, m - 1), -1);
S(1, 1) = .5 * S(1, 1);
S(end, end) = .5 * S(end, end);
S = (1 / h) * S;
end

File: tests/python_adapter/model_discrepancy/synthetic_test_with_hyperparam_auto_1D/Assemble_Mass_and_Stiffness.py
import numpy as np
import sys

class Assemble_Mass_and_Stiffness():

    def __init__(self,m):
        m = int(m)
        self.m = m
        self.x = np.linspace(0.0,1.0,m)
        h = self.x[1] - self.x[0]

        self.M = 4.0*np.eye((m))
        self.M[0,0] = 2.0
        self.M[m-1,m-1] = 2.0
        for i in range(0,m-1):
            self.M[i,i+1] = 1.0
            self.M[i+1,i] = 1.0
        self.M = (h/6.0)*self.M

        self.S = 2.0*np.eye((m))
        self.S[0,0] = 1.0
        self.S[m-1,m-1] = 1.0
        for i in range(0,m-1):
            self.S[i,i+1] = -1.0
            self.S[i+1,i] = -1.0
        self.S = (1.0/h)*self.S
